xbmc/video/VideoInfoScanner.cpp

```diff
@@ -121,36 +121,45 @@ namespace VIDEO
       // result in unexpected behaviour.
       m_bCanInterrupt = false;
 
-      bool bCancelled = false;
-      while (!bCancelled && !m_pathsToScan.empty())
+
+      if (m_handle)
+        m_handle->SetTitle("Listing media files ...");
+
+      // List all media files of all 'm_pathsToScan'
+      //   INFO: m_pathsToScan may contain path with different library content, so we store their properties into a dedicated struct.
+      std::vector<SourcePathContent> _pathsContent;
+      CFileItemList remoteDirectoryItems[m_pathsToScan.size()];
```

Comment:
Variable sized arrays are not part of the C++ standard, please replace this with a `std::vector`.